ocs_ci/deployment/ibmcloud.py

```diff
@@ -287,6 +301,100 @@ def _verify_volumes_deleted(resourece_group):
 
+    def delete_leftover_resources(self, resource_group):
+        """
+        Delete leftovers from IBM Cloud.
+
+        Args:
+            resource_group: Resource group in IBM Cloud that contains the cluster resources.
+
+        """
+
+        def _get_resources(resource_group):
+            """
+            Return a list leftover resources for the specified Resource Group
+            """
+            cmd = f"ibmcloud resource service-instances --type all -g {resource_group} --output json"
+            proc = exec_cmd(cmd)
+
+            return json.loads(proc.stdout)
+
+        def _get_reclamations(resource_group):
+            """
+            Get reclamations for resource group.
+
+            Args:
+                rsource_group (str): Resource group name
+
+            Returns:
+                list: Reclamations for resource group if found.
+            """
+            rg_id = self.get_resource_group(return_id=True)
+            cmd = "ibmcloud resource reclamations --output json"
+            proc = exec_cmd(cmd)
+            reclamations = json.loads(proc.stdout)
+            rg_reclamations = []
+            for reclamation in reclamations:
+                if reclamation["resource_group_id"] == rg_id:
+                    rg_reclamations.append(reclamation)
+            return rg_reclamations
+
+        def _delete_reclamations(reclamations):
+            """
+            Delete reclamations
+
+            Args:
+                reclamations (list): Reclamations to delete
+
+            """
+            for reclamation in reclamations:
+                logger.info(f"Deleting reclamation: {reclamation}")
+                cmd = (
+                    f"ibmcloud resource reclamation-delete {reclamation['id']} "
+                    "--comment 'Force deleting leftovers' -f"
+                )
+                exec_cmd(cmd)
+
+        def _delete_resources(resources, ignore_errors=False):
+            """
+            Deleting leftover resources.
+
+            Args:
+                resources (list): Resource leftover names.
+                ignore_errors (bool): If True, it will be ignoring errors from ibmcloud cmd.
+
+            """
+            for resource in resources:
+                logger.info(f"Deleting leftover {resource}")
+                delete_cmd = f"ibmcloud resource service-instance-delete -g {resource_group} -f --recursive {resource}"
+                if ignore_errors:
+                    try:
+                        exec_cmd(delete_cmd)
+                    except CommandFailed as ex:
+                        logger.debug(
+                            f"Exception will be ignored because ignore_error is set to true! Exception: {ex}"
+                        )
+                else:
+                    exec_cmd(delete_cmd)
+
+        if resource_group:
+            leftovers = _get_resources(resource_group)
+            if not leftovers:
+                logger.info("No leftovers found")
```

Comment:
if there is no leftovers, do we need to execute line 384 to 386 ?

Reply:
put it to else block